### ISSUE TITLE
Add the ability to specify MPN IDs when creating app registrations

### ISSUE THREAD

**Issue by SheaSmith** [microsoftgraph/msgraph-bicep-types#206]
**Is your feature request related to a problem? Please describe.**
We're currently reviewing the manual steps required for creating app registrations. We've managed to remove the vast majority of manual steps, however two prominent steps remain - setting a logo (covered by #153) and setting our MPN ID on the registration. Currently we get the following error when attempting to do this:
```json
{
  "code": "DeploymentFailed",
  "target": "/subscriptions/<id>/resourceGroups/DS-Infrastructure/providers/Microsoft.Resources/deployments/test",
  "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details.",
  "details": [
    {
      "code": "Forbidden",
      "message": "verifiedPublisher properties cannot be set during Application creation. Graph client request id: 802f3c5e-18df-46aa-85b9-cb522bf1c7a9. Graph request timestamp: 2024-12-18T03:38:24Z."
    }
  ]
}
```

with the following Bicep file:
```bicep
extension microsoftGraphV1_0

resource appReg 'Microsoft.Graph/applications@v1.0' = {
  uniqueName: 'test-registration'
  displayName: 'Test Reg'
  verifiedPublisher: {
    verifiedPublisherId: '<id>'
  }
}

```

**Describe the solution you'd like**
It would be great to be able to specify the MPN ID on our registrations in the Bicep script.

**Comment by dkershaw10** [comment]
@SheaSmith thanks for filing this issue.  Setting the verified publisher is accomplished through a separate API from /applications - https://learn.microsoft.com/graph/api/application-setverifiedpublisher?view=graph-rest-1.0&tabs=http.  This is something we were not aware of on the Graph Bicep team.

Not sure why this was separated out into a service action on a different endpoint, but it definitely represents a problem for us to expose the functionality in a declarative and idempotent fashion.

We'll need to sync with the App Platform team (who own these Graph APIs) to understand what the options are here.

**Comment by dkershaw10** [comment]
@SheaSmith **UPDATE**: we are looking into options to fix this, so that this property can be set.  In the meantime, we have updated the `verifiedPublisher` property to be read-only.

**Comment by SheaSmith** [comment]
> [@SheaSmith](https://github.com/SheaSmith) thanks for filing this issue. Setting the verified publisher is accomplished through a separate API from /applications - https://learn.microsoft.com/graph/api/application-setverifiedpublisher?view=graph-rest-1.0&tabs=http. This is something we were not aware of on the Graph Bicep team.
> 
> Not sure why this was separated out into a service action on a different endpoint, but it definitely represents a problem for us to expose the functionality in a declarative and idempotent fashion.
> 
> We'll need to sync with the App Platform team (who own these Graph APIs) to understand what the options are here.

Just tried doing this with a deploymentScript, and found another potential issue with implementing it. Currently you can only set the MPN ID via delegated permissions. So it would be possible (in theory) to do a Bicep integration for the interactive flow, but any sort of application/non-interactive flow is not possible right now

**Comment by dkershaw10** [comment]
@SheaSmith Yes - that is another challenge with this API.  Unfortunately it looks like the [publisher verification](https://learn.microsoft.com/en-us/entra/identity-platform/publisher-verification-overview#requirements) process requires a signed-in user to validate various ownership aspects before being able to link the MPN ID with the application. 

I've asked the feature owner if there's a way that the validation process could incorporate an app-only flow.
